uicomponent-compose/core/src/main/java/io/github/droidkaigi/feeder/core/animation/FadeThrough.kt

```diff
@@ -0,0 +1,126 @@
+package io.github.droidkaigi.feeder.core.animation
+
+import androidx.compose.animation.core.AnimationConstants.DefaultDurationMillis
+import androidx.compose.animation.core.Easing
+import androidx.compose.animation.core.FastOutSlowInEasing
+import androidx.compose.animation.core.MutableTransitionState
+import androidx.compose.animation.core.animateFloat
+import androidx.compose.animation.core.tween
+import androidx.compose.animation.core.updateTransition
+import androidx.compose.foundation.layout.Box
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.key
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.util.fastForEach
+
+private const val DEFAULT_START_SCALE = 0.92f
+private const val PROGRESS_THRESHOLD = 0.35f
+
+/**
+ * [FadeThrough] allows to switch between two layouts with a fade through animation.
+ *
+ * @see com.google.android.material.transition.MaterialFadeThrough
+ * @see androidx.compose.animation.Crossfade
+ *
+ * @param targetState is a key representing your target layout state. Every time you change a key
+ * the animation will be triggered. The [content] called with the old key will be faded out while
+ * the [content] called with the new key will be faded in.
+ * @param modifier Modifier to be applied to the animation container.
+ * @param durationMillis total duration of the animation.
+ * @param delayMillis the amount of time in milliseconds that animation waits before starting.
+ * @param easing the easing curve that will be used to interpolate between start and end.
+ */
+@Composable
+fun <T> FadeThrough(
```

Comment:
Did you implement this yourself??? 👀

Reply:
Yes, I refer to MaterialFadeThrough and Crossfade.